test/es6/ES6Super.js

```diff
@@ -0,0 +1,202 @@
+                }
+            }
+        }
+        A.prototype.m();
+        assert.areEqual("identifier", A.prototype.prop);
+
+        class B {
+            m() {
+                super['prop'] = "expression";
+                Object.freeze(B.prototype);
+                try {
+                    super['prop'] = "super";
+                    throw Error("Expected to throw");
+                } catch (e) {
+                    assert.areEqual("TypeError", e.name);
+                    assert.areEqual("Assignment to read-only properties is not allowed in strict mode", e.description);
+                }
+            }
+        }
+        B.prototype.m();
+        assert.areEqual("expression", B.prototype.prop);
+    }
+  },
+  {
+    name: "super property in eval",
+    body: function () {
+        var valuex, valuey, valuez, valuet;
+        
+        var a = { x: 'a', y: 'a', z: 'a', t: 'a' };
+        var b = { y: 'b', t: 'b' };
+        Object.setPrototypeOf(b, a);
+        
+        var obj = {
+            x : 'obj',
+            y : 'obj',
+            z : 'obj',
+            t : 'obj',
+            m() {
+                valuex = eval('super.x');
+                valuey = eval('super.y');
+            },
+            n() {
+                valuez = eval('super["z"]');
+                valuet = eval('super["t"]');
+            }
+        };
+        
+        Object.setPrototypeOf(obj, b);
+        obj.m();
+        assert.areEqual("a", valuex, "value x == 'a'");
+        assert.areEqual("b", valuey, "value y == 'b'");
+        obj.n();
+        assert.areEqual("a", valuez, "value z == 'a'");
+        assert.areEqual("b", valuet, "value t == 'b'");
+    }
+  },
+  {
+    name: "evaluation of expression before making super property reference",
+    body: function () {
+        var value = undefined;
+        
+        assert.throws(function(){ eval('super[value = 0]'); }, ReferenceError, "super[value = 0]", "Missing or invalid 'super' binding");
+        assert.areEqual(0, value);
+    }
+  },
+  {
+    name: "evaluation of argument list after getting super constructor",
+    body: function () {
+        var value = undefined;
+        
+        class A extends Object { constructor() { super(value = 1); } }
+        Object.setPrototypeOf(A, toString);
+        assert.throws(()=>new A(), TypeError, "Function is not a constructor", "Function is not a constructor");
+        assert.areEqual(undefined, value);
+    }
+  },
+  {
+    name: "super reference may not be deleted",
+    body: function () {
+        assert.throws(
+            function() {
+                class A extends Object {
+                    constructor() { delete super.prop; }
+                }
+                new A();
+            }, ReferenceError, "attempts to delete super property", "Unable to delete property with a super reference");
+    }
+  },
+  {
+    name: "super reference may not be deleted",

```

Comment:
Updated in latest rev.

---
In reply to: [75548069](https://github.com/Microsoft/ChakraCore/pull/1466#discussion_r75548069) [](ancestors = 75548069)